src/ORM/Table.php

```diff
@@ -1848,6 +1869,26 @@ public function delete(EntityInterface $entity, $options = [])
     }
 
     /**
+     * Try to delete an entity or throw a PersistenceFailedException if the entity is new,
+     * has no primary key value, application rules checks failed or the delete was aborted by a callback.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to remove.
+     * @param array|\ArrayAccess $options The options for the delete.
+     * @return bool success
+     * @throws \Cake\ORM\Exception\PersistenceFailedException
+     * @see \Cake\ORM\Table::delete()
+     */
+    public function deleteOrFail(EntityInterface $entity, $options = [])
+    {
+        $deleted = $this->delete($entity, $options);
+        if ($deleted === false) {
+            throw new PersistenceFailedException(['delete', $entity]);
```

Comment:
No, you're absolutely right in this case :)
So I'll overwrite the constructor and add a new `PersistenceFailedException::getEntity` method?